### PR TITLE
Disable all non authorized items in scheduler portal and resource manager

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**
Check permissions for jobs and nodes before disabling all non authorized items in scheduler portal and resource manager

### CODE REVIEW

File: common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java [hunk 1]
```diff
@@ -76,4 +143,228 @@ public String getSessionId() {
     public void setSessionId(String id) {
         this.sessionId = id;
     }
+
+    public static void addSchedulerPermissions(Map<String, Map<String, Boolean>> permissions) {//TODO OANA AICI
```
Comment (AlexFalessi) [inline common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java:147]: Forgot small // TODO at the end of line

File: common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java [hunk 2]
```diff
@@ -76,4 +143,228 @@ public String getSessionId() {
...
+
+    public boolean userHasPermissionToShutDownJob() {
+        return sessionPermissions.containsKey(PERMISSION_SCHEDULER_FRONTEND_SHUTDOWN) &&
+               sessionPermissions.get(PERMISSION_SCHEDULER_FRONTEND_SHUTDOWN);
+    }
+
+    public boolean userHasPermissionToKillJob() {
+        return sessionPermissions.containsKey(PERMISSION_SCHEDULER_FRONTEND_KILL) &&
+               sessionPermissions.get(PERMISSION_SCHEDULER_FRONTEND_KILL);
+    }
```
Comment (AlexFalessi) [inline common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java:294]: Please change method names, these actions are on the scheduler, not jobs

File: common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java [hunk 3]
```diff
@@ -76,4 +143,228 @@ public String getSessionId() {
...
+
+    public boolean userHasPermissionToLockNodes() {
+        return sessionPermissions.containsKey(PERMISSION_RM_LOCK_NODES) &&
+               sessionPermissions.get(PERMISSION_RM_LOCK_NODES);
+    }
+
+    public boolean userDoesNotHavePermissionToLockNodes() {
+        return sessionPermissions.containsKey(PERMISSION_RM_LOCK_NODES) &&
+               !sessionPermissions.get(PERMISSION_RM_LOCK_NODES);
+    }
```
Comment (AlexFalessi) [inline common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java:304]: Is it usefull to have a method for both conditions ? Cant we work with only `userHasPermissionToLockNodes()` ?

(same for others below)

File: common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java [hunk 4]
```diff
@@ -76,4 +143,228 @@ public String getSessionId() {
...
+
+    public boolean sessionPermissionWasReceivedForMethod(String method) {
+        return sessionPermissions.containsKey(method);
+    }
+
+    /**
+     * Add the user permissions for the nodes to the cashed map to avoid sending unnecessary requests
+     * @param permissions the map the node and true/false if the user has or has not the permission to the node
+     */
+    public static void addRMPermissions(Map<String, Boolean> permissions) {
```
Comment (AlexFalessi) [inline common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java:353]: I don't think this method should be static.

File: common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java [hunk 5]
```diff
@@ -76,4 +143,228 @@ public String getSessionId() {
...
+     * Check if the user has permissions to all of the given nodes
+     * @param nodeUrls the list of nodes
+     * @return true if the user has permissions to all of the given nodes
+     */
+    public boolean userHasPermissionForAllSelectedNodes(Set<String> nodeUrls) {
+        List<String> resultNodes = nodeUrls.stream()
+                                           .filter(node -> RMNodePermissions.containsKey(node) &&
+                                                           !RMNodePermissions.get(node))
+                                           .collect(Collectors.toList());
+        return resultNodes.isEmpty() ? true : false;
```
Comment (AlexFalessi) [inline common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java:367]: I think you can directly do

```
return nodeUrls.stream()
                         .allMatch(node -> RMNodePermissions.containsKey(node) && RMNodePermissions.get(node))
```

**Review by mboussaa — APPROVED** [review]

**Review by AlexFalessi — APPROVED** [review]
Please acknowledge my comments before merging

**Comment by fviale** [comment]
retest this please

**Pull request merged**